### ISSUE TITLE
Should we make entity 'type' mandatory in NGSIv2?

### ISSUE THREAD

**Issue by jmcanterafonseca** [telefonicaid/fiware-orion#1859]
even though in #1835 landed some code to assign the type 'none' to untyped entities, I'm not very happy about that. What I'm proposing is making entity type mandatory in NGSIv2, avoiding the usage of pseudo-types such as 'none' that break a bit the JSON representation. 

I would like to hear from other people what do they think about it? What's the point in having untyped entities? 

thanks

**Comment by fgalan** [comment]
In fact, having a default type used in creation/update when client is not explicetly setting it is a way of achieving what you say about making entity type mandatory from an interoperability point of view and, at the same time, support the use case of users that don't want explicit typing. What I mean by this use case is users which use the following to create

```
POST /v2/entitites
{
  "id": "myEnt",
  "attr1": 42
}
```

the following to update


```
PUT /v2/entitites/myEnt/attrs/attrs1
{
  "value": 43
}
```

and the following to retrieve

```
GET /v2/entitites
```

Why do I (as an user) have to bother with types if I don't need them to make my application work? But, at the same time, by using a default type we ensure that from a model interoperability point of view (i.e. another user updating/modifitying the same entities) all keeps consisent and that the operational requirements pointed out by @mrutid at https://github.com/telefonicaid/fiware-orion/pull/1835#issuecomment-188689164 are also fullfilled.

Note that supporting the "user don't want to bother with types" use case is important, as that case is fully supported in NGSIv1. Thus, if we wouldn't support this use case, we would be making NGSIv2 more restrictive (i.e. worse) than NGSIv1.

**Comment by jmcanterafonseca** [comment]
fair enough fine by me. You can close this issue or leave it open to know other opinions

thanks!

**Comment by fgalan** [comment]
Let's have opened for a while with a reasonable timeout if no new opinion is cast (I'm assigning the question to the 1.0.0 milestone, it will be closed at 1.0.0 releasing date).

**Comment by fgalan** [comment]
We are reaching the end of the Orion 1.0.0 work cycle and nobody has provide feedback to this issues. Thus, closing.
